src/nixpacks/builder/docker/dockerfile_generation.rs

```diff
@@ -360,16 +363,48 @@ impl DockerfileGenerator for Phase {
         };
         let phase_copy_cmd = utils::get_copy_command(&phase_files, APP_DIR);
 
-        let cache_mount = utils::get_cache_mount(&cache_key, &phase.cache_directories);
-        let cmds_str = phase
-            .cmds
-            .clone()
-            .unwrap_or_default()
+        let cmds_str = if options.incremental_cache_image.is_some() {
+            let cach_copy_in_command =
+                utils::get_copy_in_cached_dirs_command(output, &phase.cache_directories).join("\n");
+
+            let file_server_url = {
+                let this = options.file_server_url.clone();
+                let default = "http://host.docker.internal:8008/".to_string();
+                match this {
+                    Some(v) => v,
+                    None => default,
+                }
+            };
```

Comment:
This can be shortened to 

```rs
let file_server_url = options.file_server_url.clone().unwrap_or_else(|| "http://host.docker.internal:8008/".to_string())
```